subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/JvmGradlePluginProjectInitDescriptor.java

```diff
@@ -76,12 +80,12 @@ public void generateProjectBuildScript(String projectName, InitSettings settings
             BuildScriptBuilder.Expression functionalTest = buildScriptBuilder.taskRegistration("Add a task to run the functional tests", "functionalTest", "Test", b -> {
                 b.propertyAssignment(null, "testClassesDirs", buildScriptBuilder.propertyExpression(functionalTestSourceSet, "output.classesDirs"), true);
                 b.propertyAssignment(null, "classpath", buildScriptBuilder.propertyExpression(functionalTestSourceSet, "runtimeClasspath"), true);
-                if (getTestFrameworks().contains(BuildInitTestFramework.SPOCK) || getTestFrameworks().contains(BuildInitTestFramework.JUNIT_JUPITER)) {
+                if (!Sets.intersection(FRAMEWORKS_USING_JUNIT_PLATFORM, getTestFrameworks()).isEmpty()) {
                     b.methodInvocation(null, "useJUnitPlatform");
                 }
             });
             buildScriptBuilder.taskMethodInvocation("Run the functional tests as part of `check`", "check", "Task", "dependsOn", functionalTest);
-            if (getTestFrameworks().contains(BuildInitTestFramework.SPOCK) || getTestFrameworks().contains(BuildInitTestFramework.JUNIT_JUPITER)) {
+            if (!Sets.intersection(FRAMEWORKS_USING_JUNIT_PLATFORM, getTestFrameworks()).isEmpty()) {
```

Comment:
I think these conditions are always true now. 

What `getTestFrameworks()` returns is weird and it's only used here in the descriptors. The Java, Groovy and Kotlin Gradle plugin descriptors always return a single item and it's one of the frameworks from the set here. 

I would remove these conditionals. We are always going to need `useJUnitPlatform` now.